print_d.c

```diff
@@ -2,37 +2,34 @@
 #include <stdio.h>
 #include "main.h"
 
-/* printf_integer_recursive - prints a single or multi digit integer
- * @num: a single or multi digit integer
- * Return: The length of the integer
+/* printf_integer - prints a singe or multi digit integer
+ * @val: variable length argument list
+ * Return: the length of the integer
  */
-int print_integer_recursive(int num)
+int printf_integer(va_list val)
 {
+	int num = va_arg(val, int);
+	char buffer[20];
```

Comment:
Why the buffer? able to say something quick about that?

Reply:
The normal iterative method of converting an integer to a string will result in the string equivalent being in reverse order to the original integer. For example, 123 will end up as "321".
There are two options that I know to handle this, use a reverse function to print out the reversed string or use the buffer array to hold the values, then print out the buffer members from the last to the first.

Reply:
I understand this one:

- iteratively add terminal digits to increasing indices of the buffer then print it in reverse

> @zonafrank : use a reverse function to print out the reversed string

Are you saying you store the result in a buffer either way and then reverse it with a function and print out the result?

And there is the side-step that avoids a buffer, which you were using earlier: `print_integer_recursive`.

Reply:
I did not like the recursive solution because it required a helper function. I just preferred a solution in one function.
We could look at the other iterative reversal method later. It involves reversing the number twice and uses no buffer.
There might be more elegant ways, and I am keen to see them too.